gtsfm/utils/metrics.py

```diff
@@ -55,21 +62,152 @@ def count_correct_correspondences(
         Boolean mask of which verified correspondences are classified as correct under Sampson error
             (using GT epipolar geometry).
     """
-    # TODO: add unit test, with mocking.
-    if len(keypoints_i1) != len(keypoints_i2):
-        raise ValueError("Keypoints must have same counts")
+    if corr_idxs_i1i2.size == 0:
+        return None, None
+
+    # Compute ground truth correspondences.
+    matched_keypoints_i1 = keypoints_i1.extract_indices(corr_idxs_i1i2[:, 0])
+    matched_keypoints_i2 = keypoints_i2.extract_indices(corr_idxs_i1i2[:, 1])
+    if None not in [gt_scene_mesh, gt_wTi1, gt_wTi2]:
+        gt_camera_i1 = PinholeCameraCal3Bundler(gt_wTi1, intrinsics_i1)
+        gt_camera_i2 = PinholeCameraCal3Bundler(gt_wTi2, intrinsics_i2)
+        is_inlier, reproj_error = mesh_inlier_correspondences(
+            matched_keypoints_i1,
+            matched_keypoints_i2,
+            gt_camera_i1,
+            gt_camera_i2,
+            gt_scene_mesh,
+            dist_threshold,
+        )
+    elif gt_wTi1 is not None and gt_wTi2 is not None:
```

Comment:
i don't think we need this check here to see if the ground truth poses are provided since we already check it in the TwoViewEstimator:  

https://github.com/borglab/gtsfm/pull/295/files?authenticity_token=QXZc5PBgvc%2BslrvBTHXtF%2BPMIrnyqmt7Pn2kCL1%2B2atLH3gBLigzOifRNQyOrhmMLLI0OpUCYqt5D02N1VtsPw%3D%3D&file-filters%5B%5D=.py#diff-1f5b63f35ed4d88709ce154ae9808f38a1a7178fb5f74211b1053ae5a8c37894R120

Reply:
i think the double nesting here makes it harder to read, so i think a single nesting would be better, just checking to see if the mesh is there. or you can exit immediately with None,None.

We discuss this a but in [Contributing.md](https://github.com/borglab/gtsfm/blob/master/CONTRIBUTING.md#python-style), but returning early is always preferred to nesting.